test/common/http/header_map_impl_test.cc

```diff
@@ -319,15 +319,126 @@ TEST(HeaderMapImplTest, DoubleInlineAdd) {
   EXPECT_EQ(1UL, headers.size());
 }
 
+struct HeaderBrutalityResult {
```

Comment:
None of this header brutality stuff is really needed to test copying and it makes all of the tests much more confusing. I would recommend removing it all. If you want to make sure copying is working, just allocate string on the heap, add them to headers, delete the string, then access the header map. ASAN will catch any issues.